internal/runner/common/unit_resolver_dependencies.go

```diff
@@ -0,0 +1,161 @@
+package common
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/internal/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/pkg/log"
+	"github.com/gruntwork-io/terragrunt/shell"
+	"github.com/gruntwork-io/terragrunt/telemetry"
+	"github.com/gruntwork-io/terragrunt/util"
+)
+
+const maxLevelsOfRecursion = 20
+
+// Look through the dependencies of the units in the given map and resolve the "external" dependency paths listed in
+// each units config (i.e. those dependencies not in the given list of Terragrunt config canonical file paths).
+// These external dependencies are outside of the current working directory, which means they may not be part of the
+// environment the user is trying to run --all apply or run --all destroy. Therefore, this method also confirms whether the user wants
+// to actually apply those dependencies or just assume they are already applied. Note that this method will NOT fill in
+// the Dependencies field of the Unit struct (see the crosslinkDependencies method for that).
+func (r *UnitResolver) resolveExternalDependenciesForUnits(ctx context.Context, l log.Logger, unitsMap, unitsAlreadyProcessed UnitsMap, recursionLevel int) (UnitsMap, error) {
```

Comment:
What does it mean to "resolve" these? We already do a cycle check after discovery, so why are we doing this here?